query_optimizer/ExecutionGenerator.cpp

```diff
@@ -713,25 +795,73 @@ void ExecutionGenerator::convertHashJoin(const P::HashJoinPtr &physical_plan) {
   }
 
   // Create join hash table proto.
-  const QueryContext::join_hash_table_id join_hash_table_index =
-      query_context_proto_->join_hash_tables_size();
-  S::HashTable *hash_table_proto = query_context_proto_->add_join_hash_tables();
-
-  // SimplifyHashTableImplTypeProto() switches the hash table implementation
-  // from SeparateChaining to SimpleScalarSeparateChaining when there is a
-  // single scalar key type with a reversible hash function.
-  hash_table_proto->set_hash_table_impl_type(
-      SimplifyHashTableImplTypeProto(
-          HashTableImplTypeProtoFromString(FLAGS_join_hashtable_type),
-          key_types));
+  const QueryContext::join_hash_table_group_id join_hash_table_group_index =
+      query_context_proto_->join_hash_table_groups_size();
+  query_context_proto_->add_join_hash_table_groups();
+  std::vector<S::HashTable *> hash_table_proto;
+
+  bool is_numa_aware_join = false;
+  if (build_relation_info->relation->hasPartitionScheme() && probe_operator_info->relation->hasPartitionScheme()) {
+    if (build_relation_info->relation->getPartitionScheme().getPartitionSchemeHeader().getNumPartitions() ==
+        probe_operator_info->relation->getPartitionScheme().getPartitionSchemeHeader().getNumPartitions()) {
```

Comment:
If this code was in a loop this makes sense to me. I don't see the use of this here. Can you please help me understand why this would be useful in this code that is not inside a loop?